core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java

```diff
@@ -834,6 +836,40 @@ private static void writeLoadMetadata(AbsoluteTableIdentifier identifier,
     }
   }
 
+  /**
+   * Currently the segment lock files are not deleted immediately when unlock,
+   * so it needs to delete expired lock files before delete loads.
+   */
+  private static void deleteExpiredSegmentLockFiles(CarbonTable carbonTable) {
```

Comment:
move this function to CarbonTable

Reply:
It's better to move this function to CarbonLockUtil.
Done